src/test/java/com/arpnetworking/commons/math/AccumulatorTest.java

```diff
@@ -0,0 +1,215 @@
+                        1.0E10,     // magnitudes
+                        0.0001,     // signs
+                        1.1e3,      // random cancel
+                        3.9e6},     // random values
+                new Object[]{
+                        (Supplier<Accumulator>) BigDecimalAccumulator::new,
+                        0.0001,     // magnitudes
+                        0.0001,     // signs
+                        0.0001,     // random cancel
+                        0.0001},    // random values
+                new Object[]{
+                        (Supplier<Accumulator>) KahanAccumulator::new,
+                        0.0001,     // magnitudes
+                        0.0001,     // signs
+                        0.0001,     // random cancel
+                        0.5e4},     // random values
+                new Object[]{
+                        (Supplier<Accumulator>) NeumaierAccumulator::new,
+                        0.0001,     // magnitudes
+                        0.0001,     // signs
+                        0.0001,     // random cancel
+                        0.0001},    // random values
+                new Object[]{
+                        (Supplier<Accumulator>) PairwiseAccumulator::new,
+                        8.0e9,      // magnitudes
+                        3.0e16,     // signs
+                        1.1e3,      // random cancel
+                        2.0e4});    // random values
+    }
+
+    @Test
+    public void testDifferentMagnitudes() {
+        final Accumulator accumulator = _accumulatorSupplier.get();
+        accumulator.accumulate(1.0e16);
+        for (int i = 0; i < 1000000; ++i) {
+            accumulator.accumulate(1.0);
+        }
+        assertEquals(1.0000000001E16, accumulator.getSum(), _differentMagnitudesMargin);
+    }
+
+    @Test
+    public void testDifferentSigns() {
+        final Accumulator accumulator = _accumulatorSupplier.get();
+        for (int i = 0; i < 4000000; ++i) {
+            final int group = i % 4;
+            switch (group) {
+                case 0:
+                    accumulator.accumulate(1.0);
+                    break;
+                case 1:
+                    accumulator.accumulate(1.0e16);
+                    break;
+                case 2:
+                    accumulator.accumulate(-1.0);
+                    break;
+                case 3:
+                    accumulator.accumulate(-1.0e16);
+                    break;
+                default:
+                    throw new IllegalStateException("This should never happen");
+            }
+        }
+        assertEquals(0.0, accumulator.getSum(), _differentSignsMargin);
+    }
+
+    @Test
+    public void testRandomValues() {
+        final Accumulator accumulator = _accumulatorSupplier.get();
+        for (final double value : RANDOM_VALUE_DATA_SET) {
+            accumulator.accumulate(value);
+        }
+
+        // The actual value is below and is approximately 3.32e18; all the
```

Comment:
I think this comment is wrong. Looking at the sums you get I don't see how the actual value would be off by a sign. The value in the assert equals (and big decimal sum) is `-2.7135163974456428E18`

Reply:
Fixed!